docs/proposed/psa-conditional-inclusion-c.md

```diff
@@ -109,7 +109,30 @@ These symbols are not part of the public interface of Mbed TLS towards applicati
 
 #### Definition of internal inclusion symbols
 
-The header file `mbedtls/config.h` needs to define all the `MBEDTLS_xxx_C` configuration symbols, including the ones deduced from the PSA crypto configuration. It does this by including the new header file **`mbedtls/config_psa.h`**, which defines the `MBEDTLS_PSA_BUILTIN_xxx` symbols and deduces the corresponding `MBEDTLS_xxx_C` (and other) symbols.
+When `MBEDTLS_PSA_CRYPTO_CONFIG` is set, the header file `mbedtls/config.h` needs to define all the `MBEDTLS_xxx_C` configuration symbols, including the ones deduced from the PSA crypto configuration. It does this by including the new header file **`mbedtls/config_psa.h`**, which defines the `MBEDTLS_PSA_BUILTIN_xxx` symbols and deduces the corresponding `MBEDTLS_xxx_C` (and other) symbols.
+
+When `MBEDTLS_PSA_CRYPTO_CONFIG` is not set, the configuration of Mbed TLS works as before, and the inclusion of non-PSA code only depends on `MBEDTLS_xxx` symbols defined (or not) in `mbedtls/config.h`. In this case, `mbedtls/config_psa.h` is only needed to build the PSA parts of the library, including `psa/crypto_struct.h` (where the layout of some structures depends on the configuration). Therefore, `psa/crypto_struct.h` (or a header that is included before it) needs to include `mbedtls/config_psa.h`.
```

Comment:
I still don't really understand why you put some special emphasis on `psa/crypto_struct.h`. The definitions in the header depends on the configuration of the library thus it includes `mbedtls/config.h` like every .c or .h file in the project with a dependency on the configuration.  Is it the only one in `include/psa` ? No, I can see that `crypto_sizes.h` and `crypto_platform.h` include also `mbedtls/config.h`.

Regarding the case where  `MBEDTLS_PSA_CRYPTO_CONFIG` is not set, one important thing to me that is only visible in the table at the end (if I didn't miss it somewhere else) is that `mbedtls/config_psa.h` still defines `MBEDTLS_PSA_BUILTIN_xxx` symbols but based on `MBEDTLS_xxx_C` symbols not `PSA_WANT_xxx` symbols.

Reply:
Good point about `crypto_sizes.h`. The point is that it needs to be included in public headers.

Reply:
> The point is that it needs to be included in public headers.

ok we are hitting the fact that the project's header policy is not well defined here I'd say. It would be better not having to mention that, it should be obvious.

Reply:
I've reworded this paragraph, in particular to state explicitly that application code needs `PSA_WANT_xxx`. I think this should be explicit since we got it wrong at one point.